aspnet-core/src/ProjectManagement.Core/Services/ResourceService/ResourceManager.cs

```diff
@@ -1397,5 +1398,107 @@ public async Task<List<UserShortInfoDto>> GetListUserShortInfo(bool onlyActive)
+            // get data will pool
+            var listLinkedResource = await query
+                .Select(lr => new 
+                {
+                    Resource = new GetUserInfo()
+                    {
+                        Id = lr.UserId,
+                        EmailAddress = lr.User.EmailAddress,
+                        AvatarPath = lr.User.AvatarPath,
+                        UserType = lr.User.UserType,
+                        UserLevel = lr.User.UserLevel,
+                        IsActive = lr.User.IsActive,
+                        FullName = lr.User.FullName,
+                        UserName = lr.User.UserName,
+                        BranchColor = lr.User.Branch.Color,
+                        BranchDisplayName = lr.User.Branch.DisplayName,
+                        PositionColor = lr.User.Position.Color,
+                        PositionName = lr.User.Position.ShortName,
+                    },
+                    Accounts = new AccountDto()
+                    {
+                        ChargeName = lr.ProjectUserBill.AccountName ?? lr.ProjectUserBill.User.FullName,
+                        HeadCount = lr.ProjectUserBill.HeadCount,
+                        EndChargeDate = lr.ProjectUserBill.EndTime,
+                        Contribute = lr.Contribute,
+                        Note = lr.ProjectUserBill.Note
+                    }
+                })
+                .ToListAsync();
+            // group by resource
+            var groupResource = listLinkedResource
+                .GroupBy(gr => gr.Resource.Id)
+                .Select(gr => new GetAllWillPoolResourceDto()
+                {
+                    Resource = gr.First().Resource,
+                    Accounts = gr.Select(s => s.Accounts).ToList(),
+                    TotalContribute = gr.Sum(s => s.Accounts.Contribute)
+                }).ToList();
+            // sort with contribute
+            groupResource = input.SortContribute
+                ? groupResource.OrderByDescending(gr => gr.TotalContribute).ToList()
+                : groupResource.OrderBy(gr => gr.TotalContribute).ToList();
+            // sort with end charge date
+            if (input.SortEndChargeDate)
+            {
+                foreach (var group in groupResource)
+                {
+                    group.Accounts = group.Accounts
+                        .OrderByDescending(ac => ac.EndChargeDate)
+                        .ToList();
+                }
+            }
+            else
+            {
+                foreach (var group in groupResource)
+                {
+                    group.Accounts = group.Accounts
+                        .OrderBy(ac => ac.EndChargeDate)
+                        .ToList();
+                }
+            }
+            return new GridResult<GetAllWillPoolResourceDto>(groupResource, groupResource.Count);
+        }
+        
+        private static void ValidationInputFilters(InputGetAllWillPoolResourceDto input)
```

Comment:
default value này thì xly ở phía FE thôi
BE valid endChargeDatTo phải >= endChargeDateFrom là đủ

Reply:
chú ý là FE phải gửi lên dạng yyyy-MM-dd